### ISSUE TITLE
Unable to navigate tabs using arrow keys

### ISSUE THREAD

**Issue by natmasse** [IBM/carbon-components-react#1738]
## Title line template: [Title]: Brief description
Unable to navigate tabs using the arrow keys. This is an accessibility issue. A user should be able to navigate the components using a keyboard. 

> Describe in detail the issue you're having.
Using the arrow keys to navigate the Tab(s) inside a Tab component doesn't work and throws an exception (see screen capture below) 

> Is this issue related to a specific component?
Tabs / Tab 

> What did you expect to happen? What happened instead? What would you like to see changed?
Expect to be able to navigate the tabs using arrow keys

> What browser are you working in?
Chrome

> What version of the Carbon Design System are you using?
carbon-components-react v6.71.0

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
IBM cloud console
release date: end of January

## Steps to reproduce the issue

1. Add tabs to a page using Tabs and Tab carbon react components
2. Navigate to the Tabs component using tab key
3. Navigate the individual Tab(s) components using the arrow key
4. Check the error in the console 

## Additional information
<img width="477" alt="screen shot 2019-01-11 at 12 34 50 pm" src="https://user-images.githubusercontent.com/40840777/51051844-f8871b80-15a2-11e9-895b-96c790af205f.png">

**Comment by asudoh** [comment]
Hi 👋 thank you for reporting! It appears that the JavaScript error doesn't reproduce in http://react.carbondesignsystem.com/?selectedKind=Tabs&selectedStory=Default. That said, would you create a reduced case based on https://codesandbox.io/s/github/IBM/carbon-components-react/tree/master/examples/codesandbox?

**Comment by natmasse** [comment]
@asudoh I'm creating the tab(s) using the renderAnchor function to create react router Link(s). The problem is the react router Link does not have a focus() function.
In the error above "tab.tabAnchor" is the react router Link.

I've tried to overide the tab "handleTabAnchorFocus" function by passing it as a prop without any success.

**Comment by asudoh** [comment]
I see, one option can be `typeof tab.tabAnchor === 'function'` guard, though your application loses focus management feature in this way. Another approach can be done by your side, which is, providing a component with forward-ref capability to `renderAnchor`.

**Comment by natmasse** [comment]
I fixed it on our side by wrapping the react router Link in a react component. The wrapper provides the focus function using a react ref.
